From cb7c68e7b4283365d4643feba2629162c4604d14 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Wed, 13 Oct 2021 20:13:07 -0300
Subject: [PATCH 111/233] [FIO toup] imx8mq_evk: define default dfu_alt_info
 for emmc updates

Define dfu_alt_info to allow updating SPL/imx-boot and u-boot.itb via
DFU (which is also used by EFI capsule updates).

Upstream note: standard SPL seems to be called flash.bin (generated by
u-boot), so needs to be aligned there. Here we're also using the offset
and size for a mix of primary and secondary slot, so needs to be aligned
with multiboot support.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 include/configs/imx8mq_evk.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/imx8mq_evk.h b/include/configs/imx8mq_evk.h
index 027a9214cdd..fc127eedcc3 100644
--- a/include/configs/imx8mq_evk.h
+++ b/include/configs/imx8mq_evk.h
@@ -106,6 +106,8 @@
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
+	"dfu_alt_info=mmc 0=imx-boot raw 0x42 0x2be mmcpart 1;" \
+		"u-boot-itb raw 0x1300 0xD00 mmcpart 1\0" \
 	"fdtfile=imx8mq-evk.dtb\0" \
 	"bootm_size=0x10000000\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
-- 
2.25.1

